src/ORM/DataQuery.php

```diff
@@ -447,6 +448,33 @@ public function count()
         return $this->getFinalisedQuery()->count("DISTINCT {$quotedColumn}");
     }
 
+    /**
+     * Return whether this dataquery will have records. This will use `EXISTS` statements in SQL which are more
+     * performant - especially when used in combination with indexed columns (that you're filtering on)
+     *
+     * @return bool
+     */
+    public function exists()
+    {
+        // Grab a statement selecting "everything" - the engine shouldn't care what's being selected in an "EXISTS"
+        // statement anyway
+        $statement = $this->getFinalisedQuery(null, true);
+        $statement->setSelect('*');
```

Comment:
What about `$statement->setSelect('1');` ? If we're only checking existence, we don't need to select all columns.

Reply:
Good question. I'm almost certain that the query optimiser treats `*` and `1` the same when within an `EXISTS` query. The `*` is purely SQL semantics.